### PR TITLE
DDS-1821 [Button] Fix of QA test

### PR DESCRIPTION
**Pull request by rinjInTokyo**
## Related Ticket(s) and Links

- JIRA Ticket: [DDS-1821](https://daikinsvoil.atlassian.net/browse/DDS-1821)

## Problem

- Miss max width for button.
- When click link type without href button will fire click event.

## Solution

- Miss max width for button.
  - Specify max width to 100% to component host css
- When click link type without href button will fire click event.
  - When link type button without href, stop the event.

## How Has This Been Tested?

Manual confirmation (I'm not sure if we need to add a visual test case for the boundary value.)

## Checklist

- [x]  I have added a changeset file that describes the changes.
  _Hint_: run `npx changeset`.
- [x]  I have performed a self-review of my code.
- [x]  I have added/updated appropriate regression tests.
- [x]  I have confirmed that there are no accessibility warnings that can be addressed.
  _Hint_: check the Accessibility tab in the Storybook.
- [x]  I have confirmed that the bug fix does not affect the keyboard accessibility of the component.
- [x]  I have confirmed that the fix does not introduce any new visual regressions.

If you don't feel that the PR is ready for review, please create the PR in draft status.

[DDS-1821]: https://daikinsvoil.atlassian.net/browse/DDS-1821?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/button/daikin-button.ts [hunk 1]
```diff
@@ -160,7 +161,7 @@ export class DaikinButton extends LitElement {
     super();
 
     this.addEventListener("click", (event: MouseEvent): void => {
-      if (this.disabled) {
+      if (this.disabled || (this.type === "link" && this.href == undefined)) {
```
Comment (yodas7) [inline src/components/button/daikin-button.ts:164]:
```suggestion
      if (this.disabled || (this.type === "link" && this.href == null)) {
```

File: .changeset/sweet-gifts-juggle.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@daikin-oss/design-system-web-components": patch
+---
+
+Button: Updated appearance. (DDS-1821)
```
Comment (yodas7) [inline .changeset/sweet-gifts-juggle.md:5]:
```suggestion
Button: Set the maximum width by default. (DDS-1821)
Button: Fixed an issue where click events could be propagated through disabled links. (DDS-1821)
```

**Review by yodas7 — APPROVED** [review]

**Review by ukkary — APPROVED** [review]
